lib/grammar/rules/io_redirect.cpp

```diff
@@ -0,0 +1,40 @@
+#include "shell/grammar/rules/io_redirect.hpp"
+#include "shell/grammar/grammar_util.hpp"
+#include "shell/grammar/rules/io_file.hpp"
+#include "shell/grammar/rules/io_here.hpp"
+#include "shell/assert.hpp"
+
+namespace shell {
+
+Rule IORedirect::make() {
+	return Rule {
+	    IORedirect::options,
+	};
+}
+
+vector<Rule::Option> IORedirect::options() {
+	auto placeholder = [](vector<Ast::Node> &args) -> optional<Ast::Node> {
+		return nullopt;
+	};
+	auto handler = [](vector<Ast::Node> &args) -> optional<Ast::Node> {
+		LOG_DEBUG("IORedirect: handler\n");
+		if (args.size() == 2) {
+			auto &io_number_token = args[0].get<Ast::Literal>().token.get<IoNumber>();
+			auto io_number = std::stoul(io_number_token.value);
```

Comment:
handle overflow? I think catching the std::out_of_range and translating it into something like a ParseException (which can implement RecoverableError)